### ISSUE TITLE
Using PHPMailer 5.2.28 since 2 weeks I have issues to send massive mail using office365 SMTP.

### ISSUE THREAD

**Issue by gabolander** [PHPMailer/PHPMailer#2539]
Hi there, I have a php program using PHPMailer 5.2.28 to send mails to different users from 2 years ago so far. No problems using it, until almost 2 weeks ago... then I had issues. 
This is my routine to send mails using office365 SMTP : 

```
function send_a_mail($firstname, $lastname, $username, $code, $email1, $sitename, $sitetitle, $newpassword, $tries=5)
{


    $thistry=0; 
    while($thistry < $tries){    
        $adminmail="admin@mydomain.com";
        $adminname="John Smith @ MyCompany";
        
        $mail = new PHPMailer;
        $mail->SMTPDebug = 2; // 0 = off (for production use) - 1 = client messages - 2 = client and server messages
        $mail->Debugoutput = 'html';
        $mail->isSMTP();

        $mail->CharSet = 'utf-8';
        $mail->Host = gethostbyname('smtp.office365.com'); // if your network does not support SMTP over IPv6
        
        $mail->SMTPSecure = 'tls'; // ssl is depracated
        $mail->Port = 587; // TLS only
        $mail->SMTPAuth = true;
        $mail->SMTPOptions = array(
                  'ssl' => array(
                      'verify_peer' => false,
                      'verify_peer_name' => false,
                      'allow_self_signed' => true
                  )
              );        
        
        
        $mail->Username = 'user@mycompany.com';
        $mail->Password = 'MyPazzvord';
        $mail->setFrom("user@mycompany.com", "Service IT");
        $mail->addAddress($email1,$firstname." ".$lastname);
        $mail->addReplyTo($adminmail, $adminname);
        $mail->Subject = "Object $sitetitle";
        $dispcode=($code)?"(Codice $code)":"";
        
        $message_text="Bla bla bla";
        $message_html="Bla bla bla by $sitename <br>";  

        $mail->msgHTML($message_html); 

        if(!$mail->send()){
            $ret=-1;
            echo "ERRORE MAIL su HOST:".$mail->Host." verso $firstname [$email1]: " . $mail->ErrorInfo . "\n";
            sleep(2);
        }else{
            echo "Message sent to $firstname [$email1]!";
            $ret=0;
            $thistry=$tries; // set to the max to quit the while loop
        }
        $thistry++;

        
        unset($mail); // object destroy
    }
    return $ret;
}

```

The error that occurs **is not systematic**. When it happens, in _$mail->ErrorInfo_ I have "_SMTP Connect() failed_" and if I look Debug output, I read something like "_SMTP NOTICE: EOF caught while checking if connected_" ... 
If you look at the code, I did 5 tries to send the mail. Sometimes happens that at third or forth try I can send successfully the mail, sometimes it sends at first tries and other times it fails all the 5 tries without sending the mail... 
I don't know if I have to add some options to make the code more robust and reliable, but I swear that until a pair of week or so, it always worked on the first shot like a charm. 
Maybe Microsoft changed something on their servers? I didn't try PHPMailer 6.x yet, because I should change a lot of code in my infrastructure to make its usage in my procedures ... but if you say that it's the only thing I can do to avoid troubles, I will do it though ... 
Any other suggestions?
    
Thanks a lot in advance.
Gabo

**Comment by xdumont** [comment]
The problem is on Microsoft side. We get a notification this morning:
![image](https://user-images.githubusercontent.com/14092929/139641681-85fe4067-650c-4176-94d3-a5e8c969dee9.png)

**Comment by Synchro** [comment]
I've seen some other complaints about this that have disabled certificate verification, which leads me to think that it's not a verification problem, but a protocol mismatch. Older versions of PHP may support only TLS 1.0 and 1.1, and I suspect that Microsoft has been doing a rolling update of their servers that drops support for those obsolete protocol versions, just as they have for HTTP over the last year. This also explains why it's intermittent, as they will have many thousands of serves, some of which will still accept old protocols, but others won't. Since you're running PHPMailer 5.2, the only reason for doing that is because you're using an obsolete version of PHP too, so that's probably the cause.

In short, update your PHP version, preferably to 8.0 since 7.4 is end-of-life this month.

**Comment by Synchro** [comment]
@xdumont apologies, I didn't read your comment properly. It's good to have confirmation that this was indeed the problem, though TBH, I'm a bit disappointed that it wasn't the start of a permanent Microsoft policy to get rid of TLS 1.0! Maybe they were just practicing – we can but hope!

**Comment by xdumont** [comment]
@Synchro , no problem ;-). For sure, staying on a old release of PHP is not a good plan for the future. I think TLS1.0 support will be removed in a near future as already annonce by MS (but delayed because of COVID).
Last week, we fix this problem by implementing the SMTP Service on our MS domain server and relay email to Microsoft. This use TLS 1.2 between this SMTP server and MS365.
